ThermoPower/Gas.mo

```diff
@@ -2742,6 +2742,83 @@ The packages Medium are redeclared and a mass balance determines the composition
 </html>"));
   end FanMech;
 
+  package Utility "Utility models"
+
+    model ClosedSystemInit
+      "Component for the steady-state initialization of closed systems"
+      replaceable package Medium = Modelica.Media.Interfaces.PartialMedium
+        "Medium model"
+        annotation(choicesAllMatching = true);
+      parameter Medium.AbsolutePressure pstart "Start value of pressure";
+      parameter Boolean initEquationFromInput = false
+        "Get initial condition residual from input connector"
+        annotation(Dialog(group="External inputs"), choices(checkBox=true));
+      parameter Boolean useHomotopy
+        "Use homotopy for residual-based initialization";
+      final parameter Medium.MassFlowRate w_b(fixed = false, start = 0)
+        "Initial balance flow rate eventually equal to zero";
+      outer ThermoPower.System system "System wide properties";
+
+      FlangeB flange(redeclare package Medium = Medium, m_flow(min = 0))
+         annotation (Placement(transformation(extent={{-20,-100},{20,-60}},
+                               rotation=0),
+                               iconTransformation(extent={{-20,-100},{20,-60}})));
+      Modelica.Blocks.Interfaces.RealInput initEquationResidual if initEquationFromInput
+        "Residual of initial equation" annotation (Placement(transformation(extent={
+                {-60,0},{-20,40}}), iconTransformation(extent={{-60,-10},{-40,10}})));
+    protected
+      Modelica.Blocks.Interfaces.RealInput initEquationResidual_int
+        "Hidden internal connector for conditional connection";
+    equation
+      0 = flange.m_flow + w_b "Mass balance";
+      flange.h_outflow = Medium.h_default

```

Comment:
I guess you should also assign flange.Xi_outflow, otherwise there are missing equations in case of gas mixtures